ui/src/app/factory/InterceptedHttp.ts

```diff
@@ -4,19 +4,34 @@ import {Injectable} from '@angular/core';
 import {ConnectionBackend, Http, Request, RequestOptions, RequestOptionsArgs,
   Response} from '@angular/http';
 import {ErrorHandlingService} from 'app/services/error-handling.service';
-import {Observable} from 'rxjs/Rx';
+import {Observable, Subject} from 'rxjs/Rx';
 
 @Injectable()
 export class InterceptedHttp extends Http {
+  private statusSubject = new Subject<boolean>();
+  public shouldPingStatus = true;
+
 
   constructor(private backend: ConnectionBackend, private defaultOptions: RequestOptions,
       private errorHandlingService: ErrorHandlingService) {
     super(backend, defaultOptions);
   }
 
   request(url: string | Request, options?: RequestOptionsArgs): Observable<Response> {
-    return this.errorHandlingService.retryApi(
+    const response = this.errorHandlingService.retryApi(
         super.request(url, options));
+    response.subscribe(() => {}, (e) => {
```

Comment:
Since http requests are not fired until the observable is subscribed to, I'm curious how this interacts with our api calls - does this double every request?  If so, you may want to change this `.subscribe` to just a `.catch` and return _that_ so that execution remains deferred.  I haven't tried that, but it looks like that's how rxjs `catch` works; you can attach an error handler to the chain that will catch anything upstream of it without causing evaluation.

Reply:
I don't think this actually changes anything, because the request won't be called unless someone is already making an API call, in which case I would expect there to be a subscribe. (I believe?)

That said, catch may be more what we want. I will investigate.